uint8 TYPE_ABOVE = 0
uint8 TYPE_ALIGNED = 1
uint8 TYPE_AT = 2
uint8 TYPE_COLLIDED = 3
uint8 TYPE_HOVERED = 4
uint8 TYPE_IN = 5
uint8 TYPE_LOW_BATTERY = 6
uint8 TYPE_SCANNED = 7

uint8 type
PredicateAbove above
PredicateAligned aligned
PredicateAt at
PredicateCollided collided
PredicateHovered hovered
PredicateIn in
PredicateLowBattery low_battery
PredicateScanned scanned

bool truth_value
